projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -118,10 +149,54 @@ export class TableWidgetRendererComponent
     );
   }
 
+  public onFilterChange(item: ToggleItem<TableWidgetFilterModel>): void {
+    if (item.value && item.value.attribute && item.value.operator && item.value.value) {
+      this.toggleFilterSubject.next([
+        {
+          field: item.value.attribute,
+          operator: item.value.operator,
+          value: item.value.value
+        }
+      ]);
+
+      return;
+    }
+
+    this.toggleFilterSubject.next([]);
+  }
+
+  public onSearchChange(text: string): void {
+    const searchFilter: TableFilter = {
+      field: this.api.model.searchAttribute!,
+      operator: FilterOperator.Like,
+      value: text
+    };
+    this.searchFilterSubject.next([searchFilter]);
+  }
+
+  public onModeChange(mode: TableMode): void {
```

Comment:
We should never be writing back to the model - please store it locally for the current state, the model should be the persisted default.

Reply:
Fixed